pkg/utils/build/quay.go

```diff
@@ -158,3 +191,36 @@ func GetImageTag(organization, repository, tagName string) (quay.Tag, error) {
 		}
 	}
 }
+
+func ValidateMediaTypeInImage(imageUrl, mediaType string) (bool, error) {
+
+	cmd := exec.Command("skopeo", "inspect", "docker://"+imageUrl) // #nosec G204
+	fmt.Printf("Running command: %q\n", cmd.String())
+	outputBytes, err := cmd.Output()
+	if err != nil {
+		return false, fmt.Errorf("error while running skopeo cmd: %v", err)
+	}
+	skopeoInspectOutput := ImageInspectInfo{}
+	err = json.Unmarshal(outputBytes, &skopeoInspectOutput)
+	if err != nil {
+		return false, fmt.Errorf("error while unmarshalling skopeo cmd output: %v", err)
+	}
+
+	if mediaType == "oci" {
+		for _, layerData := range skopeoInspectOutput.LayersData {
+			if layerData.MIMEType != mimeTypeOci {
```

Comment:
I don't think that we need to check the layers themselves. I think we can just check the `mediaType` for the manifest. For OCI, the mediaType would be `application/vnd.oci.image.manifest.v1+json`. For the docker format, it should be `application/vnd.docker.distribution.manifest.v2+json`.

https://github.com/opencontainers/image-spec/blob/main/media-types.md

Reply:
thanks for the feedback, updated the PR to check the manifest mediaType, PTAL again